spec/common/misc/utils.spec.ts

```diff
@@ -20,20 +20,84 @@ describe('Utils Service', () => {
         });
 
         it('should handle valid urls', () => {
-            expect(Utils.getDomain('https://bitwarden')).toBe('bitwarden');
-            expect(Utils.getDomain('https://bitwarden.com')).toBe('bitwarden.com');
+            expect(Utils.getDomain('bitwarden.com')).toBe('bitwarden.com');
             expect(Utils.getDomain('http://bitwarden.com')).toBe('bitwarden.com');
+            expect(Utils.getDomain('https://bitwarden.com')).toBe('bitwarden.com');
+
+            expect(Utils.getDomain('www.bitwarden.com')).toBe('bitwarden.com');
+            expect(Utils.getDomain('http://www.bitwarden.com')).toBe('bitwarden.com');
+            expect(Utils.getDomain('https://www.bitwarden.com')).toBe('bitwarden.com');
+
+            expect(Utils.getDomain('vault.bitwarden.com')).toBe('bitwarden.com');
             expect(Utils.getDomain('http://vault.bitwarden.com')).toBe('bitwarden.com');
+            expect(Utils.getDomain('https://vault.bitwarden.com')).toBe('bitwarden.com');
+
+            expect(Utils.getDomain('www.vault.bitwarden.com')).toBe('bitwarden.com');
+            expect(Utils.getDomain('http://www.vault.bitwarden.com')).toBe('bitwarden.com');
+            expect(Utils.getDomain('https://www.vault.bitwarden.com')).toBe('bitwarden.com');
+
+            expect(Utils.getDomain('user:password@bitwarden.com:8080/password/sites?and&query#hash'))
+                .toBe('bitwarden.com');
+            expect(Utils.getDomain('http://user:password@bitwarden.com:8080/password/sites?and&query#hash'))
+                .toBe('bitwarden.com');
             expect(Utils.getDomain('https://user:password@bitwarden.com:8080/password/sites?and&query#hash'))
                 .toBe('bitwarden.com');
+
+            expect(Utils.getDomain('bitwarden.unknown')).toBe('bitwarden.unknown');
+            expect(Utils.getDomain('http://bitwarden.unknown')).toBe('bitwarden.unknown');
             expect(Utils.getDomain('https://bitwarden.unknown')).toBe('bitwarden.unknown');
         });
 
-        it('should support localhost and IP', () => {
+        it('should handle valid urls with an underscore in subdomain', () => {
+            expect(Utils.getDomain('my_vault.bitwarden.com/')).toBe('bitwarden.com');
+            expect(Utils.getDomain('http://my_vault.bitwarden.com/')).toBe('bitwarden.com');
+            expect(Utils.getDomain('https://my_vault.bitwarden.com/')).toBe('bitwarden.com');
+        });
+
+        it('should support urls containing umlauts', () => {
```

Comment:
I don't think these are valid test cases because domain names do not *actually* contain unicode characters. If we want the domain name from a string containing unicode, I think it should do the punycode translation in order to get the "real" domain name. e.g.

`expect(Utils.getDomain('bütwarden.com')).toBe('xn--btwarden-65a.com');`

However, this does depend on what components/services are using this method. There may be some situations where we want to display the unicode in the GUI. But for behind the scenes domain matching logic, we always need the punycode. I suggest doing some investigation here.